rio_tiler/tasks.py

```diff
@@ -51,10 +64,12 @@ def create_tasks(reader: Callable, assets, threads, *args, **kwargs) -> TaskType
     """Create Future Tasks."""
     if threads and threads > 1:
         with futures.ThreadPoolExecutor(max_workers=threads) as executor:
-            return [
-                (executor.submit(reader, asset, *args, **kwargs), asset)
-                for asset in assets
-            ]
+            return iter(
```

Comment:
wrapping the futures inside an iterator to be able to use `next`

Reply:
You might just be able to use a generator comprehension instead? Replace the `[ ]` with parentheses?

Reply:
tried and it didn't work 🤷‍♂️

Reply:
(I think because the future needs to be executed with the excetutor. if you use a generator, the code will be run outide (in the filer function)

Reply:
That makes sense 👍